### PR TITLE
Dimensional tear update

### PR DESCRIPTION
**Pull request by Miraviel**
## What Does This PR Do
Please be patient, this is my first bigger PRs. :c

This change intends to update the Dimensional Rift random event, to make it a little more than someone injecting a gold slime with plasma.

Design document behind it:

### Abstract

The Dimensional Rift event spawns some monsters after a short warning period, creating fast-and-harmful (but not round-ending) issues in a small place. It also breaks nearby consoles to add some minor engineering issues to the mayhem.

### Goals

1) To generate **medium to high severity medical problems**, by spawning hard-hitting but not permakilling monsters; to ensure medical has something to do, even on slow/extended rounds
2) To generate **low-level engineering issues** by breaking computer screens, to give engineering/assistants some minor work to do, without completely nuking a workplace should they be absent
3) To keep this **event rather short**
4) To give ghosts an early heads up about something interesting happening
5) To make an otherwise simple mob-spawning event stand out by making it **spooky**!

### Non-goals

1) To completely take out the area in which it spawns
2) To make it so deadly it no longer fits the "medium severity" random event group
3) About the spooky atmosphere: breaking lights may sound logical to do, but shrouding lethal mobs in darkness is not fun gameplay-wise, so this idea was scrapped

### Content

A rift appears randomly on the station, does the following:
* **Boos nearby machines** as an early warning; lore-wise, the area is getting affected by unstable alien energy, causing the electrical devices to go nuts.
* After a few seconds, a **portal spawns** (with a new sound effect!) and **breaks nearby computers and mirrors**. Why mirrors? Because they bring bad luck, of course!
* Some **hostile monsters spawn** with a "leader" mob.
  * There are three types: **faithless, creature, and mi-go**. Faithless is a low-hitting (15 dmg), low health (80) monster, basically a "thank god this guy spawned instead of the others". Creature/mi-go are hard-hitting (25-50), low health (80) monsters, you can melee them, but they will hurt you too before dying.
  * The leader mob is a hard-hitting (30-50), high health (300) modified **lesser hellhound**, that is guaranteed to ruin someone's day if they approach it in melee. It is supposed to be either swarmed or killed by guns.
* To avoid visual noise, the portal deletes itself fairly soon.

### Potential Changes

* **Damage/health** values of the monsters.
  * The Hellhound should stay hard-hitting, tanky mob so it takes more than a random assistant bumrushing it with a welder, albeit I am open for soft-tuning any of the four monsters.
* The **number of the monsters**.
  * To my knowledge, we cannot accurately track how many active personnel we have on the station currently. If this could be done, we could adjust the number of spawns according to the number of active players.

## Why It's Good For The Game
1) An otherwise simple mob-spawning event has some atmosphere to it.
2) The current one ends rather soon if two, welding tool-wielding assistants happened to be nearby.
3) It made no sense that a tear in the fabric resulted in bears and bats spawning. Bring out the spooky monsters!
4) With this, we can use the fancy hellhound mob.
5) A new audio cue for the impending doom near you!

## Images of changes

Video with audio - **UPDATED**
https://streamable.com/cn3x6u

## Changelog
:cl:
add: The Dimensional Rift random event has been updated, it is now spookier and somewhat deadlier than before.
/:cl:

### CODE REVIEW

File: code/modules/events/tear.dm [hunk 1]
```diff
@@ -1,47 +1,101 @@
...
+
+// We spawn a leader mob to make the portal actually dangerous.
+/obj/effect/tear/proc/spawn_leader()
+	var/mob/living/simple_animal/M = new /mob/living/simple_animal/hostile/hellhound(get_turf(src))
+	M.name = "frenzied hellhound" 	// Rename it to indicate that their stats are different from the regular one.
+	M.melee_damage_lower = 30		// Heavy melee damage to encourage the usage of guns.
+	M.melee_damage_upper = 50
+	M.maxHealth = 300				// The minions are low health, hard hitting creatures; the leader should stand till the end.
+	M.health = 300
+	M.faction = list("rift")
```
Comment (hal9000PR) [inline code/modules/events/tear.dm:99]: please make a subtype of the mob and new that instead.

File: code/modules/events/tear.dm [hunk 2]
```diff
@@ -1,47 +1,101 @@
...
-	name="Dimensional Tear"
-	desc="A tear in the dimensional fabric of space and time."
-	icon='icons/effects/tear.dmi'
-	icon_state="tear"
+	name = "dimensional tear"
+	desc = "A tear in the dimensional fabric of space and time."
+	icon = 'icons/effects/tear.dmi'
+	icon_state = "tear"
 	density = 0
 	anchored = 1
```
Comment (Vi3trice) [inline code/modules/events/tear.dm:46]:
```suggestion
```
Effects have no density and are anchored by default.

File: code/modules/events/tear.dm (hunk 1 above)
Comment (Miraviel) [reply code/modules/events/tear.dm]: done in 43f256270113cd432155f337a7faeaa855fdcbea

File: code/modules/events/tear.dm (hunk 2 above)
Comment (Miraviel) [reply code/modules/events/tear.dm]: done in 7ad6cfa0fa04efb8038a83189fe47d49e885ad27
I did not even notice this old code, thank you!

**Comment by Miraviel** [comment]
As per @Cocacolagua 's request, instead of flickering consoles, all machinery is boo'd for `maximum spook`.

Video updated accordingly.

**Comment by Miraviel** [comment]
Linter is angy and I'm not sure how to fix it.

Fails here: `Error: failed to resolve path /obj/machinery/get_spooked`

If I change `INVOKE_ASYNC(M, /obj/machinery/get_spooked)` to `INVOKE_ASYNC(M, /obj/machinery/.proc/get_spooked)`, then I get this error in dreammaker:
`code\modules\events\tear.dm:27:error: /obj/machinery/.proc/get_spooked: undefined type path`

File: code/modules/events/tear.dm [hunk 3]
```diff
@@ -1,47 +1,93 @@
...
+		// Give ghosts some time to jump there before it begins.
+		var/image/alert_overlay = image('icons/mob/animal.dmi', "hellhound")
+		notify_ghosts("\A [src] is about to open in [get_area(T)].", title = "Dimensional Rift", source = T, alert_overlay = alert_overlay, action = NOTIFY_FOLLOW)
+		addtimer(CALLBACK(src, .proc/spawn_tear, T), 4 SECONDS)
 
-/datum/event/tear/setup()
-	impact_area = findEventArea()
+		// Energy overload; we mess with machines as an early warning and for extra spookiness.
+		for(var/obj/machinery/M in range(8, T))
+			INVOKE_ASYNC(M, /obj/machinery/get_spooked)
```
Comment (AffectedArc07) [inline code/modules/events/tear.dm:27]:
```suggestion
			INVOKE_ASYNC(M, .proc/get_spooked)
```
Try this. I remember having an issue with invoke async and the linter a while ago

Comment (Miraviel) [reply code/modules/events/tear.dm:27]: Sadly, this returns an error:
`code\modules\events\tear.dm:27:error: .get_spooked: undefined type path`

Comment (Miraviel) [reply code/modules/events/tear.dm]: I had to call the parent proc with `INVOKE_ASYNC(M, /atom/.proc/get_spooked)`, the linter is now satisfied

**Review by AffectedArc07 — APPROVED** [review]

**Comment by Sikroth** [comment]
Just a small observation that I remember seeing in the terror spiders event, while it sounds like far from an accurate measurement, there seems to be at least a way to check the number of connected clients, with `length(GLOB.clients)`.

The [terrors event uses it to lock the strongest/weakest spawns to lower and higher pops](https://github.com/ParadiseSS13/Paradise/blob/b0ca2e366ac12c9d106255d2f584a3b441afd27d/code/modules/events/spider_terror.dm#L25). Not sure if that's worth the trouble but if you wanted scaling mob spawns I guess it's worth looking into?

**Pull request merged**